mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -533,6 +545,21 @@ impl Nexus {
     }
 
     async fn on_rebuild_complete(&mut self, task: String) -> Result<(), Error> {
+        match RebuildTask::lookup(&task) {
+            Ok(t) => {
+                if t.state == RebuildState::Paused {
+                    // Leave all states as they are
+                    return Ok(());
+                }
+            }
+            Err(_) => {
```

Comment:
lookup already returns a rebuild task not found error. If you make RebuildOperationError a generic error then you can skip the match

Reply:
Done